### DISCUSSION TITLE
Strange 2-Minute Lag When Opening Connection

### DISCUSSION THREAD

**Discussion by pha3z** [PHPMailer/PHPMailer#2873] [General]
This is totally weird.

Debug output shows "Opening to smtp.gmail.com:587" followed by an enormous 2-minute wait before proceeding to the handshake.

Look at the Timestamps:
21:30:00  and 21:32:11

How could this be possible??

`2023-02-16 **21:30:00** [2@ip-172-16-100-44]

3 - Connection: opening to smtp.gmail.com:587, timeout=300, options=array (
  'ssl' =>
  array (
    'verify_peer' => false,
    'verify_peer_name' => false,
    'allow_self_signed' => true,
  ),
)

2023-02-16 **21:32:11** [2@ip-172-16-100-44]

3 - Connection: opened2023-02-16 21:32:11 [2@ip-172-16-100-44]

2 - SERVER -> CLIENT: 220 smtp.gmail.com ESMTP b186-20020a3780c3000000b0072c86374ddfsm2455610qkd.71 - gsmtp

2023-02-16 21:32:11 [2@ip-172-16-100-44]

1 - CLIENT -> SERVER: EHLO reliableparts.zirrus.one

2023-02-16 21:32:11 [2@ip-172-16-100-44]

2 - SERVER -> CLIENT: 250-smtp.gmail.com at your service, [54.163.72.86]
...
`

**Comment by pha3z** [comment]
SOLVED:
PHPMailer was attempting to connect on IPv6, which would eventually timeout, and then it would fallback to IPv4.

**Reply by Synchro** [reply]
Good that you figured that out. Generally speaking, SMTP is not good for interactive use (e.g. during web page processing) because it does have some really long timeouts, several of which are routinely used by receivers for greetdelay, throttling, greylisting, etc.

When connecting to somewhere well-known like gmail, you should never disable certificate verification. There is lots in the docs about that.
